src/tools/droplr/droplruploader.cpp

```diff
@@ -0,0 +1,206 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// SPDX-FileCopyrightText: 2017-2019 Alejandro Sirgo Rica & Contributors
+
+#include "droplruploader.h"
+#include "src/utils/confighandler.h"
+#include "src/utils/filenamehandler.h"
+#include "src/utils/history.h"
+#include "src/utils/systemnotification.h"
+#include "src/widgets/imagelabel.h"
+#include "src/widgets/loadspinner.h"
+#include "src/widgets/notificationwidget.h"
+#include <QApplication>
+#include <QBuffer>
+#include <QClipboard>
+#include <QCursor>
+#include <QDesktopServices>
+#include <QDrag>
+#include <QGuiApplication>
+#include <QHBoxLayout>
+#include <QJsonDocument>
+#include <QJsonObject>
+#include <QLabel>
+#include <QMimeData>
+#include <QNetworkAccessManager>
+#include <QNetworkReply>
+#include <QNetworkRequest>
+#include <QPushButton>
+#include <QRect>
+#include <QScreen>
+#include <QShortcut>
+#include <QTimer>
+#include <QUrlQuery>
+#include <QVBoxLayout>
+
+DroplrUploader::DroplrUploader(const QPixmap& capture, QWidget* parent)
+  : QWidget(parent)
+  , m_pixmap(capture)
+{
+    setWindowTitle(tr("Upload to Imgur"));
+    setWindowIcon(QIcon(":img/app/flameshot.svg"));
+
+#if (QT_VERSION >= QT_VERSION_CHECK(5, 10, 0))
+    QRect position = frameGeometry();
+    QScreen* screen = QGuiApplication::screenAt(QCursor::pos());
+    position.moveCenter(screen->availableGeometry().center());
+    move(position.topLeft());
+#endif
+
+    m_spinner = new LoadSpinner(this);
+    m_spinner->setColor(ConfigHandler().uiMainColorValue());
+    m_spinner->start();
+
+    m_infoLabel = new QLabel(tr("Uploading Image"));
+
+    m_vLayout = new QVBoxLayout();
+    setLayout(m_vLayout);
+    m_vLayout->addWidget(m_spinner, 0, Qt::AlignHCenter);
+    m_vLayout->addWidget(m_infoLabel);
+
+    m_NetworkAM = new QNetworkAccessManager(this);
+    connect(m_NetworkAM,
+            &QNetworkAccessManager::finished,
+            this,
+            &DroplrUploader::handleReply);
+
+    setAttribute(Qt::WA_DeleteOnClose);
+
+    upload();
+    // QTimer::singleShot(2000, this, &DroplrUploader::onUploadOk); // testing
```

Comment:
Should this be removed if it was only for testing?

Reply:
@mmahmoudian 
Actually I copied imgur.cpp file and this was the part of that file, I will certainly remove it from code. :)